### PR TITLE
Add support for toplevel array and unique_ptr in GetMissingDictionary

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

File: core/meta/src/TClass.cxx [hunk 1]
```diff
@@ -4002,6 +4002,11 @@ void TClass::GetMissingDictionaries(THashTable& result, bool recurse)
       return;
    }
 
+   if (strncmp(fName, "unique_ptr<", 11) == 0 || strncmp(fName, "array<", 6) == 0) {
```
Comment (Axel-Naumann) [inline core/meta/src/TClass.cxx:4005]: And `tuple<`?

Comment (pcanal) [reply core/meta/src/TClass.cxx:4005]: humm .. good point ...

Comment (pcanal) [reply core/meta/src/TClass.cxx]: added

Comment (vgvassilev) [reply core/meta/src/TClass.cxx]: Those magic strings should probably be moved in a central place as we have them in various places already.

**Comment by pcanal** [comment]
@phsft-bot build just on mac1014/python3

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**